src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -9,7 +9,7 @@ import com.typesafe.scalalogging.LazyLogging
 import io.grpc.Status.Code
```

Comment:
My general strategy in `ClusterMonitorActor` is to:
- make `QueryForCluster ` query both cluster AND instance status from Google, and map to the appropriate action based on (google cluster status, google instance status, Leo cluster status).
- make `NotReadyCluster`, `ReadyCluster`, `StoppedCluster`, `FailedCluster` persist the current state of instances to the DB. That way the instances are naturally updated as the cluster is monitored.

It seems to work. Still, the logic here feels a little hairy - thinking about ways to make it better in the future.